src/Microsoft.SqlTools.SqlCore/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs

```diff
@@ -1,356 +1,353 @@
+                }
+
+                childAsMeItem.NodeSubType = GetNodeSubType(context, smoContext);
+                childAsMeItem.NodeStatus = GetNodeStatus(context, smoContext);
+            }
+        }
+
+        internal virtual Type[] ChildQuerierTypes
+        {
+            get
+            {
+                return null;
+            }
+        }
+
+        public override IEnumerable<INodeFilter> Filters
+        {
+            get
+            {
+                return Enumerable.Empty<INodeFilter>();
+            }
+        }
+
+        public override IEnumerable<NodeSmoProperty> SmoProperties
+        {
+            get
+            {
+                return Enumerable.Empty<NodeSmoProperty>();
+            }
+        }
+
+        internal IEnumerable<NodeSmoProperty> CachedSmoProperties
+        {
+            get
+            {
+                return smoProperties == null ? SmoProperties : smoProperties;
+            }
+        }
+
+        protected virtual Type[] GetChildQuerierTypes(TreeNode parent)
+        {
+            return ChildQuerierTypes;
+        }
+
+        /// <summary>
+        /// Returns true if any final validation of the object to be added passes, and false
+        /// if validation fails. This provides a chance to filter specific items out of a list 
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <param name="contextObject"></param>
+        /// <returns>boolean</returns>
+        public virtual bool PassesFinalFilters(TreeNode parent, object context)
+        {
+            return true;
+        }
+
+        public override string GetNodeSubType(object smoObject, SmoQueryContext smoContext)
+        {
+            return string.Empty;
+        }
+
+        public override string GetNodeStatus(object smoObject, SmoQueryContext smoContext)
+        {
+            return string.Empty;
+        }
+
+        public static bool IsPropertySupported(string propertyName, SmoQueryContext context, NamedSmoObject smoObj, IEnumerable<NodeSmoProperty> supportedProperties)
+        {
+            var property = supportedProperties.FirstOrDefault(x => string.Compare(x.Name, propertyName, StringComparison.InvariantCultureIgnoreCase) == 0);
+            if (property != null)
+            {
+                return ServerVersionHelper.IsValidFor(context.ValidFor, property.ValidFor);
+            }
+            else
+            {
+                // Return true if cannot find the proeprty, SMO still tries to get that property but adding the property to supported list can make loading the nodes faster
+                Logger.Verbose($"Smo property name {propertyName} for Smo type {smoObj.GetType()} is not added as supported properties. This can cause the performance of loading the OE nodes");
+                return true;
+            }
+        }
+
+        public override string GetNodeCustomName(object smoObject, SmoQueryContext smoContext)
```

Comment:
While you are here, can you fix all Public API Docs?

Reply:
Maybe do that as a separate PR so they can actually be reviewed easily...